explorer/bench/Bench/Pos/Explorer/ServerBench.hs

```diff
@@ -3,222 +3,124 @@ module Bench.Pos.Explorer.ServerBench
-
-    slotStart     <- liftIO $ generate $ arbitrary
-
-    blockHHs      <- liftIO $ generate $ arbitrary
-    blundsFHHs    <- liftIO $ generate $ withDefConfigurations arbitrary
-
-    slotLeaders   <- produceSlotLeaders blocksNumber
-    secretKeys    <- produceSecretKeys blocksNumber
-
-    blocks <- withDefConfigurations $
-        produceBlocksByBlockNumberAndSlots blocksNumber slotsPerEpoch slotLeaders secretKeys
-
-    let tipBlock = Prelude.last blocks
-
-    pure $ GeneratedTestArguments
-        { _gtaPageNumber              = Just pageNumber
-        , _gtaTipBlock                = tipBlock
-        , _gtaBlockHeaderHashes       = blockHHs
-        , _gtaBlundsFromHeaderHashes  = blundsFHHs
-        , _gtaSlotStart               = slotStart
-        , _gtaSlotLeaders             = slotLeaders
-        }
 
 ----------------------------------------------------------------
 -- Mocked functions
 ----------------------------------------------------------------
 
-getBlocksTotalMock
-    :: (HasConfigurations)
-    => GeneratedTestArguments
-    -> IO Integer
-getBlocksTotalMock genTestArgs = do
-    testParams <- testParamsGen
-
-    -- We replace the "real" database with our custom data.
-    let mode :: ExplorerMockableMode
-        mode = def { emmGetTipBlock = pure $ genTestArgs ^. gtaTipBlock }
-
-    -- The extra context so we can mock the functions.
-    let extraContext :: ExtraContext
-        extraContext = makeMockExtraCtx mode
-
-    runExplorerTestMode testParams extraContext getBlocksTotal
+type BenchmarkTestParams = (ExplorerTestParams, ExtraContext)
 
+-- | @getBlocksTotal@ function for benchmarks.
+getBlocksTotalBench
+    :: BenchmarkTestParams
+    -> IO Integer
+getBlocksTotalBench (testParams, extraContext) =
+    withDefConfigurations $
+        runExplorerTestMode testParams extraContext getBlocksTotal
 
--- | The actual benched function.
-getBlocksPageMock
-    :: (HasConfigurations)
-    => GeneratedTestArguments
+-- | @getBlocksPage@ function for the last page for benchmarks.
+getBlocksPageBench
+    :: BenchmarkTestParams
     -> IO (Integer, [CBlockEntry])
-getBlocksPageMock genTestArgs = do
-    testParams <- testParamsGen -- TODO(ks): Temporary test params, will be removed.
-
-        -- We replace the "real" database with our custom data.
-    let mode :: ExplorerMockableMode
-        mode = def {
-            emmGetTipBlock         = pure $ genTestArgs ^. gtaTipBlock,
-            emmGetPageBlocks       = \_ -> pure $ Just $ genTestArgs ^. gtaBlockHeaderHashes,
-            emmGetBlundFromHH      = \_ -> pure $ Just $ genTestArgs ^. gtaBlundsFromHeaderHashes,
-            emmGetSlotStart        = \_ -> pure $ Just $ genTestArgs ^. gtaSlotStart,
-            emmGetLeadersFromEpoch = \_ -> pure $ Just $ genTestArgs ^. gtaSlotLeaders
-        }
+getBlocksPageBench (testParams, extraContext) =
+    withDefConfigurations $
+        runExplorerTestMode testParams extraContext $ getBlocksPage Nothing (Just 10)
```

Comment:
Could we use `Just defaultPageSize` here?

Reply:
Fixed.